crates/oxc_linter/src/rules/typescript/prefer_ts_expect_error.rs

```diff
@@ -0,0 +1,229 @@
+pub struct PreferTsExpectError;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Enforce using @ts-expect-error over @ts-ignore.
+    ///
+    /// ### Why is this bad?
+    /// TypeScript allows you to suppress all errors on a line by placing a comment starting with @ts-ignore or @ts-expect-error immediately before the erroring line.
+    /// The two directives work the same, except @ts-expect-error causes a type error if placed before a line that's not erroring in the first place.
+    ///
+    /// This means it's easy for @ts-ignores to be forgotten about, and remain in code even after the error they were suppressing is fixed.
+    /// This is dangerous, as if a new error arises on that line it'll be suppressed by the forgotten about @ts-ignore, and so be missed.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // @ts-ignore
+    /// const str: string = 1;
+    ///
+    /// /**
+    /// * Explaining comment
+    /// *
+    /// * @ts-ignore */
+    /// const multiLine: number = 'value';
+    /// ```
+    PreferTsExpectError,
+    correctness
+);
+
+impl Rule for PreferTsExpectError {
+    fn run_once(&self, ctx: &LintContext) {
+        let comments = ctx.semantic().trivias().comments();
+
+        for (start, &comment) in comments {
+            let raw = &ctx.semantic().source_text()[*start as usize..comment.end() as usize];
+
+            if !is_valid_ts_ignore_present(comment, raw) {
+                continue;
+            }
+
+            if comment.is_single_line() {
+                let comment_span = Span::new(*start - 2, comment.end());
+                ctx.diagnostic_with_fix(PreferTsExpectErrorDiagnostic(comment_span), || {
+                    Fix::new(
+                        format!("//{}", raw.replace("@ts-ignore", "@ts-expect-error")),
+                        comment_span,
+                    )
+                });
+            } else {
+                let comment_span = Span::new(*start - 2, comment.end() + 2);
+                ctx.diagnostic_with_fix(PreferTsExpectErrorDiagnostic(comment_span), || {
+                    Fix::new(
+                        format!("/*{}*/", raw.replace("@ts-ignore", "@ts-expect-error")),
+                        comment_span,
+                    )
+                });
+            }
+        }
+    }
+}
+
+fn get_last_comment_line(comment: Comment, raw: &str) -> String {
+    if comment.is_single_line() {
+        return String::from(raw);
+    }
+
+    return String::from(raw.lines().last().unwrap_or(raw));
+}
+
+fn is_valid_ts_ignore_present(comment: Comment, raw: &str) -> bool {
+    let line = get_last_comment_line(comment, raw);
+
+    if comment.is_single_line() {
+        test_single_line_comment(&line)
+    } else {
+        test_multi_line_comment(&line)
+    }
+}
+
+fn test_single_line_comment(line: &str) -> bool {
+    lazy_static! {
+        static ref TS_IGNORE_SINGLE_LINE_REGEX: Regex = Regex::new(r"^\s*/?\s*@ts-ignore").unwrap();
```

Comment:
I think it's trivial enough to rewrite this regex with string methods. A regex is quite heavy.